ember-core/shared/src/test/scala/org/http4s/ember/core/h2/H2StreamSuite.scala

```diff
@@ -106,6 +147,71 @@ class H2StreamSuite extends Http4sSuite {
     } yield ()
   }
 
+  test(
+    "client should not hang when endStream is sent by H2Frame.Headers as trailers"
```

Comment:
You can consult RFC7540 8.1.3 for more details.

https://httpwg.org/specs/rfc7540.html#rfc.section.8.1.3